### DISCUSSION TITLE
Are there some running examples in production?

### DISCUSSION THREAD

**Discussion by biapar** [Dotnet-Boxed/Templates#624] [Q&A]
### Question

Are there some running examples in production?

**Comment by RehanSaeed** [comment]
When I had an MVC website template a while back, I found about 100 examples of sites using my template online (They just deployed the default template and then clicked a link back to my blog). I checked some later and found real sites.

With API's it's harder to find them online in an automated fashion. I've used my API template. At least one person at Microsoft has contacted me to tell me they use the API template. I know @VictorioBerra has deployed a site. Many others have also done so but they don't come and tell me about it.

Hope that helps.

**Comment by VictorioBerra** [comment]
I'm running multiple in prod at work but it's behind our firewall. Did you have any specific questions?

**Comment by biapar** [comment]
I wish to build a 3 layer tracker realtime application with vue.js as UI - web api - sql db on .net core with Active Directory authentication. Is there also a multitier support?

**Comment by VictorioBerra** [comment]
So, the template can be customized to do whatever you want it to. It wont solve all of those just out of the box. For example, if you want SQL DB you have to get the relevant nuget packages, you have wire up the EF dbcontext or dapper or whatever you want to use, and then you have to add your own layers like a repository if thats what you want.

Its important to know this template just configures core to follow best practices like https by default, and support for docker and azure, and some security config defaults for Kestrel, Swagger for APIs which is another best practice.

if you want anything more, or anything less, you have to add it. For example you want N-Tier so you may want to remove some of[ the sample code showing CQRS](https://github.com/Dotnet-Boxed/Templates/blob/master/Source/ApiTemplate/Source/ApiTemplate/Commands/). Which is why I think templates are not always the best thing for someone new to ASPNET Core or the language.

Although, I will say I have built a similar production app with almost exactly the items you listed (except a different UI framework than Vue) and it went great.

**Comment by biapar** [comment]
I see that you have 3 templates. Which Do you suggest to start from?

**Comment by VictorioBerra** [comment]
In this repo there are 4. You should learn what graphql is and what Microsoft Orleans is and then decide which one works best for you. It sounds to me like what you described above you should go with the API template and then customise it to your needs. `dotnet new api`
